bundles/go/go_fmt.moon

```diff
@@ -0,0 +1,58 @@
+import app, config, mode, signal from howl
+import Process from howl.io
+
+append = table.insert
+
+run_command = (file) ->
+  args = {}
+  for arg in config.go_fmt_command\gmatch '%S+'
+    append args, arg
+  append args, file
+  success, out, err, p = pcall Process.execute, args
+  unless success
+    log.error out
+    return
+  unless p.successful
+    log.error err
+    return
+
+reload_buffer = (buffer) ->
+  buffer.text = buffer.file.contents
+  buffer.sync_etag = buffer.file.etag
+  buffer.modified = false
+  
+calculate_new_pos = (pos, before, after) ->
+  new_pos = 1
+  i = 1
+  while i <= pos and new_pos <= #after
+    if before[i] == after[new_pos]
+      i += 1
+      new_pos += 1
+    elseif before[i]\match '%s'
+      i += 1
+    else
+      new_pos += 1
+  new_pos-1
+
+fmt = (buffer) ->
+  return unless buffer.mode.name == 'go'
+
+  file = buffer.file
+  before = file.contents
+  run_command file
```

Comment:
Ah, I see. From the code it looks like signals run synchronously only until the first coroutine yield happens (which would happen when running an external process). Did the earlier implementation have a race condition too? Say if you start typing immediately after 'save' and 'gofmt' takes longer to run, the buffer contents would be switched out, discarding your edits?

A quick solution would be run goftm on 'buffer-saved' and immediately save it again. This of course has the same race as above.

Given we might want to do similar transforms on save for other modes, maybe a better solution is warranted. @nilnor - any ideas?